pkg/registry/core/rest/storage_core.go

```diff
@@ -224,6 +230,9 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generi
 	if legacyscheme.Registry.IsEnabledVersion(schema.GroupVersion{Group: "policy", Version: "v1beta1"}) {
 		restStorageMap["pods/eviction"] = podStorage.Eviction
 	}
+	if c.IDTokenIssuer != nil && utilfeature.DefaultFeatureGate.Enabled(features.TokenRequest) {
```

Comment:
would prefer checking if `serviceAccountStorage.Token` is nil here (and making it be nil if you pass in a nil IDTokenIssuer to REST construction)